requirements-dev.txt

```diff
@@ -1,20 +1,79 @@
-autoflake
-coverage
-black
-bump2version
-codespell
-flake8
-isort
-mypy
-pip
-pydocstyle
-pylint
-pytest
-pytest-runner
-pyyaml
-sphinx
-sphinx-autodoc-typehints
-sphinx-rtd-theme
-tox
-twine
-wheel
+alabaster==0.7.12
+appdirs==1.4.4
+astroid==2.5.1
+attrs==20.3.0
+autoflake==1.4
+Babel==2.9.0
+black==20.8b1
+bleach==3.3.0
+bump2version==1.0.1
+certifi==2020.12.5
+cffi==1.14.5
+chardet==4.0.0
+click==7.1.2
+codespell==2.0.0
+colorama==0.4.4
+coverage==5.5
+cryptography==3.4.6
+distlib==0.3.1
+docutils==0.16
+filelock==3.0.12
+flake8==3.9.0
+idna==2.10
+imagesize==1.2.0
+importlib-metadata==3.7.3
+iniconfig==1.1.1
+isort==5.7.0
+jeepney==0.6.0
+Jinja2==2.11.3
+keyring==23.0.0
+lazy-object-proxy==1.5.2
+MarkupSafe==1.1.1
+mccabe==0.6.1
+mypy==0.812
+mypy-extensions==0.4.3
+packaging==20.9
+pathspec==0.8.1
+pkginfo==1.7.0
+pluggy==0.13.1
+py==1.10.0
+pycodestyle==2.7.0
+pycparser==2.20
+pydocstyle==5.1.1
+pyflakes==2.3.0
+Pygments==2.8.1
+pylint==2.7.2
+pyparsing==2.4.7
+pytest==6.2.2
+pytest-runner==5.3.0
+pytest-subprocess==1.0.0
+pytz==2021.1
+PyYAML==5.4.1
+readme-renderer==29.0
+regex==2020.11.13
+requests==2.25.1
+requests-toolbelt==0.9.1
+rfc3986==1.4.0
+SecretStorage==3.3.1
+six==1.15.0
+snowballstemmer==2.1.0
+Sphinx==3.5.2
+sphinx-autodoc-typehints==1.11.1
+sphinx-rtd-theme==0.5.1
+sphinxcontrib-applehelp==1.0.2
+sphinxcontrib-devhelp==1.0.2
+sphinxcontrib-htmlhelp==1.0.3
+sphinxcontrib-jsmath==1.0.1
+sphinxcontrib-qthelp==1.0.3
+sphinxcontrib-serializinghtml==1.1.4
+toml==0.10.2
+tox==3.23.0
+tqdm==4.59.0
+twine==3.4.1
+typed-ast==1.4.2
+typing-extensions==3.7.4.3
+urllib3==1.26.4
+virtualenv==20.4.3
+webencodings==0.5.1
+wrapt==1.12.1
+zipp==3.4.1
```

Comment:
Let's talk about this. I don't like it.

My main problem is: how do you ever get out from these fixed dependencies versions and evolve?

Reply:
I reverted the requirements freeze for now. I agree we need a better approach.  I've seen some projects maintain two lists, one with the actual requirements unpinned, another with the frozen.